catalog/glue/schema.go

```diff
@@ -0,0 +1,107 @@
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package glue
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/apache/iceberg-go"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/glue/types"
+)
+
+// schemaToGlueColumns converts an Iceberg schema to a list of Glue columns.
+func schemaToGlueColumns(schema *iceberg.Schema) []types.Column {
+	var columns []types.Column
+	for _, field := range schema.Fields() {
+		columns = append(columns, fieldToGlueColumn(field))
+	}
+
+	return columns
+}
+
+// fieldToGlueColumn converts an Iceberg nested field to a Glue column.
+func fieldToGlueColumn(field iceberg.NestedField) types.Column {
+	column := types.Column{
+		Name:    aws.String(field.Name),
+		Comment: aws.String(field.Doc),
+	}
+	column.Type = aws.String(icebergTypeToGlueType(field.Type))
+
+	return column
+}
+
+// icebergTypeToGlueType converts an Iceberg type to a Glue type string representation.
+// It handles primitive types as well as nested types like structs, lists, and maps.
+func icebergTypeToGlueType(typ iceberg.Type) string {
+	switch t := typ.(type) {
+	case iceberg.BooleanType:
+		return "boolean"
+	case iceberg.Int32Type:
+		return "int"
+	case iceberg.Int64Type:
+		return "bigint"
+	case iceberg.Float32Type:
+		return "float"
+	case iceberg.Float64Type:
+		return "double"
+	case iceberg.DateType:
+		return "date"
+	case iceberg.TimeType:
+		return "string" // Glue doesn't have a time type. TODO: Double check if this is correct
+	case iceberg.TimestampType:
+		return "timestamp"
+	case iceberg.TimestampTzType:
+		return "timestamp" // TODO: Double check if this is correct
+	case iceberg.StringType:
+		return "string"
+	case iceberg.UUIDType:
+		return "string" // Represent UUID as string
+	case iceberg.BinaryType:
+		return "binary"
+	case iceberg.DecimalType:
+		return fmt.Sprintf("decimal(%d,%d)", t.Precision(), t.Scale())
+	case iceberg.FixedType:
+		return fmt.Sprintf("binary(%d)", t.Len())
+	case *iceberg.StructType:
+		// For struct types, create a struct<field1:type1,field2:type2,...> representation
+		var fieldStrings []string
+		for _, field := range t.Fields() {
+			fieldStrings = append(fieldStrings,
+				fmt.Sprintf("%s:%s", field.Name, icebergTypeToGlueType(field.Type)))
+		}
+
+		return fmt.Sprintf("struct<%s>", strings.Join(fieldStrings, ","))
+	case *iceberg.ListType:
+		// For list types, create an array<type> representation
+		elementField := t.ElementField()
+
+		return fmt.Sprintf("array<%s>", icebergTypeToGlueType(elementField.Type))
+	case *iceberg.MapType:
+		// For map types, create a map<keyType,valueType> representation
+		keyField := t.KeyField()
+		valueField := t.ValueField()
+
+		return fmt.Sprintf("map<%s,%s>",
+			icebergTypeToGlueType(keyField.Type),
+			icebergTypeToGlueType(valueField.Type))
```

Comment:
is there documentation we can link to for Glue that specifies these formats?

Reply:
https://docs.aws.amazon.com/glue/latest/dg/glue-types.html#glue-types-catalog
Here you go. I have added the link to the documentation in the `icebergTypeToGlueType` method. It also points to Hive documentation https://cwiki.apache.org/confluence/display/hive/languagemanual+types